### PR TITLE
Add Bend Fun types

### PR DESCRIPTION
**Pull request by developedby**

### CODE REVIEW

File: Bend/Fun/Type/Type.agda [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  Var  : String → Type
+  Arr  : Type → Type → Type
+  Ctr  : String → List Type → Type
+  Tup  : List Type → Type
+  U24  : Type
+  I24  : Type
+  F24  : Type
+  None : Type
+  Number : Type → Type
+  Integer : Type → Type
```
Comment (enricozb) [inline Bend/Fun/Type/Type.agda:18]: Colon alignment `:`

File: Bend/Fun/Book/Type.agda [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+open import Bend.Fun.Term.Type
+
+private
+  import Bend.Fun.FnDef.Type as FnDef'
+  open module FnDef = FnDef' Term
+
+record Book : Set where
+  constructor MkBook
+  field
+    defs : Map FnDef
```
Comment (enricozb) [inline Bend/Fun/Book/Type.agda:13]: pls terminate in newlinew

File: Bend/Fun/TextLocation/Type.agda [hunk 3]
```diff
@@ -0,0 +1,9 @@
+module Bend.Fun.TextLocation.Type where
+
+open import Data.Nat.Type
+
+record TextLocation : Set where
+  constructor MkTextLocation
+  field
+    line : Nat
+    char : Nat
```
Comment (enricozb) [inline Bend/Fun/TextLocation/Type.agda:9]: I would consider sharing this across the `Fun` and `Imp`.

**Review by enricozb — APPROVED** [review]

File: Bend/Fun/TextLocation/Type.agda (hunk 3 above)
Comment (developedby) [reply Bend/Fun/TextLocation/Type.agda]: good idea

**Review by enricozb — APPROVED** [review]
looks good

**Pull request merged**